docs/products/clickhouse/concepts/service-architecture.rst

```diff
@@ -0,0 +1,99 @@
+
+Deployment modes
+----------------
+
+Aiven for ClickHouse can be deployed either as a single node, a single shard of three nodes, or multiple shards of three nodes each.
+
+* With a single shard, all data is present in all servers and all servers can be used for reads and writes (no main server, leader, or replica).
+* With multiple shards, the data is split between all shards and the data of each shard is present in all nodes of the shard.
+
+Each Aiven for ClickHouse service is exposed as a single server URL pointing to all servers with connections going randomly to any of the servers. ClickHouse is responsible for replicating the writes between the servers.
+For synchronizing critical low-volume information between servers, Aiven for ClickHouse relies on :ref:`ZooKeeper <zookeeper>`, which runs on each ClickHouse server.
+
+Coordinating services
+---------------------
+
+Each Aiven for Clickhouse node runs Clickhouse, ZooKeeper, and Astacus.
+
+.. _zookeeper:
+
+ZooKeeper
+'''''''''
+
+ZooKeeper is responsible for the cross-nodes coordination and synchronization of the following replication processes:
+
+* Replication of database changes across the cluster: CREATE, UPDATE, or ALTER TABLE (by ClickHouse's ``Replicated`` :ref:`database engine <replicated-database-engine>`)
+* Replication of table data across the cluster (by ClickHouse's ``ReplicatedMergeTree`` :ref:`table engine <replicated-table-engine>`). Data itself is not written to ZooKeeper but transferred directly between ClickHouse servers.
+* Replication of the storage of Users, Roles, Quotas, Row Policies for the whole cluster (by ClickHouse).
+
+  .. note::
+
+    Storing entities such as Users, Roles, Quotas, and Row Policies in ZooKeeper ensures that Role Based Access Control (RBAC) is applied consistently over the entire cluster. This type of entity storage was developed at Aiven and is now part of the upstream ClickHouse.
+
+ZooKeeper handles one process per node and is accessible only from within the cluster.
+
+.. _astacus:
+
+Astacus
+'''''''
+
+Astacus is an open-source project originated at Aiven for coordinating backups of cluster databases, including ClickHouse.
+
+Data architecture
+-----------------
+
+Aiven for ClickHouse enforces
+
+* Full schema replication: all databases, tables, users, and grants are the same on all nodes.
+* Full data replication: table rows are the same on all nodes within a shard.
+
+:ref:`Astacus <astacus>` does most of the backup and restore operations.
+
+Engines: database and table
+---------------------------
+
+ClickHouse has engines in two flavors: table engines and database engines.
+
+* Database engine is responsible for manipulating tables and decides what happens when you try to list, create, or delete a table. It can also be used to restrict a database to specific table engine or to manage replication.
```

Comment:
```suggestion
* The database engine is responsible for manipulating tables and decides what happens when you try to list, create, or delete a table. It can also be used to restrict a database to specific table engine or to manage replication.
```

Reply:
It's a good practice to omit articles at the beginning of each list item (see e.g. https://www.nngroup.com/articles/presenting-bulleted-lists/#:~:text=4.%20Avoid%20repeating%20the%20same%20word(s)%20at%20the%20beginning%20of%20each%20list%20item.).